src/renderer/screens/manager/AppsList/AppsList.js

```diff
@@ -110,10 +113,22 @@ const AppsList = ({
       installQueue,
       type: ["installed"],
     },
-    sort,
+    { type: "default", order: "asc" },
   );
 
-  const displayedAppList = onDeviceTab ? installedAppList : appList;
+  const distributionOrder = distribution.apps.map(({ name }) => name);
+
+  const displayedAppList = onDeviceTab
+    ? installedAppList
+        .sort(
+          ({ name: _name }, { name }) =>
+            distributionOrder.indexOf(_name) > distributionOrder.indexOf(name) ? -1 : 0, // order by distribution in device
+        )
+        .sort(
+          ({ name: _name }, { name }) =>
+            installQueue.indexOf(_name) > installQueue.indexOf(name) ? -1 : 0, // place install queue on top of list
+        )
```

Comment:
twice doing a sort and with indexOf each time is probably not going to be very performant.